runtime_scan/pkg/provider/aws/config.go

```diff
@@ -15,40 +15,89 @@
 
 package aws
 
-import "github.com/spf13/viper"
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+)
 
 const (
-	AWSSubnetID            = "AWS_SUBNET_ID"
-	AWSJobImageID          = "AWS_JOB_IMAGE_ID"
-	AWSSecurityGroupID     = "AWS_SECURITY_GROUP_ID"
-	AWSInstanceType        = "AWS_INSTANCE_TYPE"
-	defaultAWSJobImageID   = "ami-0568773882d492fc8" // ubuntu server 22.04 LTS (HVM), SSD volume type
-	defaultAWSInstanceType = "t2.large"
+	EnvScannerRegion       = "VMCLARITY_AWS_SCANNER_REGION"
+	EnvSubnetID            = "VMCLARITY_AWS_SUBNET_ID"
+	EnvSecurityGroupID     = "VMCLARITY_AWS_SECURITY_GROUP_ID"
+	EnvKeyPairName         = "VMCLARITY_AWS_KEYPAIR_NAME"
+	EnvScannerImage        = "VMCLARITY_AWS_SCANNER_AMI_ID"
+	EnvScannerInstanceType = "VMCLARITY_AWS_SCANNER_INSTANCE_TYPE"
+	EnvBlockDeviceName     = "VMCLARITY_AWS_BLOCK_DEVICE_NAME"
+
+	DefaultScannerInstanceType = "t2.large"
+	DefaultBlockDeviceName     = "xvdh"
 )
 
 type Config struct {
-	AmiID           string // image id of a scanner job
-	SubnetID        string // the scanner's subnet ID
-	SecurityGroupID string // the scanner's security group
-	InstanceType    string // the scanner's instance type
+	// Region where the Scanner instance needs to be created
+	ScannerRegion string
+	// SubnetID where the Scanner instance needs to be created
+	SubnetID string
+	// SecurityGroupID which needs to be attached to the Scanner instance
+	SecurityGroupID string
+	// KeyPairName is the name of the SSH KeyPair to use for Scanner instance launch
+	KeyPairName string
+	// ScannerImage is the AMI image used for creating Scanner instance
+	ScannerImage string
+	// ScannerInstanceType is the instance type used for Scanner instance
+	ScannerInstanceType string
+	// BlockDeviceName contains the block device name used for attaching Scanner volume to the Scanner instance
+	BlockDeviceName string
 }
 
-func setConfigDefaults() {
-	viper.SetDefault(AWSJobImageID, defaultAWSJobImageID)
-	viper.SetDefault(AWSInstanceType, defaultAWSInstanceType)
+func (c *Config) Validate() error {
+	if c.ScannerRegion == "" {
+		return fmt.Errorf("parameter Region must not be nil")
+	}
+
+	if c.SubnetID == "" {
+		return fmt.Errorf("parameter SubnetID must not be nil")
+	}
+
+	if c.SecurityGroupID == "" {
+		return fmt.Errorf("parameter SecurityGroupID must not be nil")
+	}
+
+	if c.ScannerImage == "" {
+		return fmt.Errorf("parameter ScannerImage must not be nil")
+	}
+
+	if c.ScannerInstanceType == "" {
+		return fmt.Errorf("parameter ScannerInstanceType must not be nil")
+	}
 
-	viper.AutomaticEnv()
+	return nil
 }
 
-func LoadConfig() *Config {
-	setConfigDefaults()
+func NewConfig() (*Config, error) {
+	// Avoid modifying the global instance
+	v := viper.New()
+
+	v.AllowEmptyEnv(true)
+	v.AutomaticEnv()
```

Comment:
I don't think we need AutomaticEnv if we're using BindEnv explicitly. Alternatively we could switch to:
```
v.SetEnvPrefix("VMCLARITY_AWS")
v.AutomaticEnv()
```
and avoid the need to define the environment variables and use BindEnv entirely. Only difference would be that `VMCLARITY_AWS_SCANNER_REGION` would become `VMCLARITY_AWS_SCANNERREGION` because of rule used by viper is:

> It will check for an environment variable with a name matching the key uppercased and prefixed with the EnvPrefix if set.

As we're now using Unmarshal, adding a new configuration option would then be as simple as adding a new item o the struct, no need to change anywhere else.

Reply:
I am aware of this, but to make the Env var readably, we need a `StringReplacer` which I am thinking about to add, so there will be no need for defining env vars, but will be automatically generated by `viper`. The downside of this that it is not straightforward to ppl who are not familiar with `viper`.

Reply:
No need for a string replacer or anything custom, viper will just use the ToUppered struct key name + the prefix from SetEnvPrefix('VMCLARITY_AWS') so if the struct looks like:

```
struct {
  Foo string
}
```

then viper will look for `VMCLARITY_AWS_FOO` in the environment. This is all natively supported by viper.

As I said the only downside is that `VMCLARITY_AWS_SCANNER_REGION` would become `VMCLARITY_AWS_SCANNERREGION` because the struct looks like:

```
struct {
 ScannerRegion string
}
```

so `ScannerRegion` becomes `SCANNERREGION`.

I'd need to double check but viper also uses mapstructure tags, so it might be possible to do:

```
struct {
  ScannerRegion string `map_structure:'scanner_region'`
}
```

and keep the underscore, need to experiment though.

Reply:
> As I said the only downside is that VMCLARITY_AWS_SCANNER_REGION would become VMCLARITY_AWS_SCANNERREGION because the struct looks like.

Thats the exact reason for me want to use a `StringReplacer` as the generated env vars would be hard to read.
Will add a `StringReplacer` in the next update.

Reply:
Also `StringReplacer` is part of the public API while the fact that `viper` uses `mapstructure` under the hood is an implementation detail which we should not directly rely on.

Reply:
mapstructure tags are part of viper's public API, they call it out in their README in the unmarshalling section: https://github.com/spf13/viper#unmarshaling

Reply:
Alright. Whatever.

Reply:
Updated the code to use `mapstructure` struct tags for  unmarshaling the `Config` struct, but it turned out that it is not working. It seems there is a bug related to this: https://github.com/spf13/viper/issues/1545
@sagikazarmark

Reply:
Right...try setting a default for the value in Viper.

Reply:
@sambetts-cisco Managed to make it work. Updated the PR.

Reply:
awesome, this looks way better from my perspective than having to maintain the environment variable constants. It also means that if we decide to use a yaml or json config, or even cobra flags we're already set up for it.